js/ag_reference.js

```diff
@@ -171,6 +174,17 @@ AttributeGroupReference.prototype = {
          return this._uri;
      },
 
+     /**
+      * This will generate a new unfiltered reference each time.
+      * Returns a reference that points to all entities of current table
+      *
+      * @type {ERMrest.Reference}
+      */
+     get unfilteredReference() {
+         var newLocation = new AttributeGroupLocation(this.location.service, this.location.catalogId, [module._fixedEncodeURIComponent(this.table.schema.name),module._fixedEncodeURIComponent(this.table.name)].join(":"));
```

Comment:
Add a check to verify that `this.table` is defined. If it doesn't exist you can either return `null`, or throw an appropriate error (if you just call verify function that'll do it).

Reply:
I updated all places in the code to include a table in the constructor for `AttributeGroupReference` so if one of these references is missing the property and `unfilteredReference` is called, I think it should throw an error. 

There are 2 places in the tests that I didn't add the table to the constructor where I could test the error case of this function.